### PR TITLE
 feat(block, panel, pick-list-group, tip, tip-manager): Add 'headingLevel' property

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** https://github.com/Esri/calcite-components/issues/1531

## Summary

feat(block, panel, pick-list-group, tip, tip-manager): Add 'headingLevel' property to specify a number at which section headings should start for the component.

- Adds a `headingLevel` prop to any components that use a `h1`-`h6`.
- Uses `headingLevel` prop and a functional component to return the correct heading level
- Allows a user to change the level for headings within a component for accessibility and semantics.

## TODO

- ~~@asangma we will need to set sizing for all the heading classes so that changing the level does not change the size of the heading.~~ We can tackle these as the arise or we can create an issue.
- ~~If naming is approved, I will create tests.~~ done

### CODE REVIEW

**Comment by driskull** [comment]
@jcfranco can you review?

**Comment by driskull** [comment]
Ping @jcfranco @paulcpederson

**Review by paulcpederson — APPROVED** [review]

**Comment by paulcpederson** [comment]
LGTM

File: src/components/functional/HeadingWizard.tsx
Comment (jcfranco) [inline src/components/functional/HeadingWizard.tsx:11]: Despite my love for wizardry, why not name this `Heading`?

File: src/components/functional/HeadingWizard.tsx
Comment (jcfranco) [inline src/components/functional/HeadingWizard.tsx:16]: Should this also have something for non-heading elements that support `aria-level`? https://www.w3.org/TR/html-aria/

File: src/components/calcite-pick-list-group/calcite-pick-list-group.tsx
Comment (jcfranco) [inline src/components/calcite-pick-list-group/calcite-pick-list-group.tsx:31]: Would it be possible to set `level` on the parent and have it adjust and pass it to their children?

File: src/components/calcite-panel/calcite-panel.tsx
Comment (jcfranco) [inline src/components/calcite-panel/calcite-panel.tsx:68]: Do you think we would need to expose the level as a shared resource? The only use cases I can think of right now is for testing and possibly keeping levels in sync between components.

**Review by jcfranco — APPROVED** [review]
Looking good! 😎 👍 

Can you add E2E tests for the level defaults before merging? Other than that, I just had a few suggestions.

**Comment by macandcheese** [comment]
Awesome add, as long as the styles are somewhat heavy handed and all headings result in the same visual display. Nice!

File: src/components/calcite-pick-list-group/calcite-pick-list-group.tsx
Comment (driskull) [reply src/components/calcite-pick-list-group/calcite-pick-list-group.tsx:31]: We could, but that would add some complexity. I would say to hold off on this for now. 

For example, the `calcite-tip-manager` could add the `heading-level` to all of it's children `calcite-tip`s pretty easily since it already has an observer to query those. The `calcite-pick-list-group` is a little more complex because I think it can be nested so it would have to figure out it's heading from its parent.

Maybe separate issues for these?

File: src/components/functional/HeadingWizard.tsx
Comment (driskull) [reply src/components/functional/HeadingWizard.tsx:16]: @jcfranco I don't think we want to support using `role="heading"` unless we have a good use case for that. Seems like we would always just be semantic?

Comment (driskull) [reply src/components/functional/HeadingWizard.tsx:16]: @paulcpederson I did find one heading role here: https://github.com/Esri/calcite-components/blob/master/src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.tsx#L119-L121

Can we just use a real heading here? Was there a good reason to use the role?

Comment (driskull) [reply src/components/functional/HeadingWizard.tsx:16]: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role#best_practices

File: src/components/calcite-pick-list-group/calcite-pick-list-group.tsx
Comment (driskull) [reply src/components/calcite-pick-list-group/calcite-pick-list-group.tsx:31]: I'll take care of the tips one now but can you create an issue for the pick-list-group?

**Comment by driskull** [comment]
I can haz merge? :)

File: src/components/calcite-pick-list-group/calcite-pick-list-group.tsx
Comment (jcfranco) [reply src/components/calcite-pick-list-group/calcite-pick-list-group.tsx]: Sure, I'll create follow-ups later today.

**Comment by jcfranco** [comment]
You may merge when ready...

when-ready

**Pull request merged**
